met-web/src/components/layout/Header/InternalHeader.tsx

```diff
@@ -1,104 +1,424 @@
+
+    useEffect(() => {
+        if (onStateChange) onStateChange(tenantDrawerOpen);
+    }, [tenantDrawerOpen, onStateChange]);
+
+    if (noOtherTenants) {
+        return (
+            <Link to="/home">
+                <BodyText sx={{ userSelect: 'none' }}>{currentTenant.name}</BodyText>
+            </Link>
+        );
+    }
+
+    const tenantList = tenants.map((tenant) => (
+        <MenuItem
+            key={tenant.short_name}
+            component={Link}
+            href={`/${tenant.short_name}/home`}
+            sx={{
+                paddingLeft: 2,
+                paddingRight: 2,
+                fontSize: '16px',
+                textTransform: 'capitalize',
+            }}
+        >
+            {tenant.name}
+        </MenuItem>
+    ));
+
+    if (shouldUseMobileView) {
+        return (
+            <>
+                <ButtonBase
+                    aria-haspopup
+                    aria-controls={tenantDrawerOpen ? 'tenant-drawer' : undefined}
+                    aria-expanded={tenantDrawerOpen}
+                    aria-label={'Tenant Switcher'}
+                    onClick={() => {
+                        setTenantDrawerOpen(!tenantDrawerOpen);
+                    }}
+                    sx={{
+                        height: 'calc( 100% - 2px )' /* 1px on either side to show border*/,
+                        ...dropdownMenuStyles,
+                    }}
+                >
+                    <TenantButtonContent isOpen={tenantDrawerOpen} />
+                </ButtonBase>
+                <Drawer
+                    anchor="top"
+                    open={tenantDrawerOpen}
+                    onClose={() => {
+                        setTenantDrawerOpen(false);
+                    }}
+                    sx={{
+                        zIndex: (theme: Theme) => theme.zIndex.drawer + 3, // render under app bar but above side nav
+                        '& .MuiDrawer-paper': {
+                            padding: '1rem',
+                            top: '6.5rem',
+                            backgroundImage: 'none',
+                            borderBottomRightRadius: '16px',
+                        },
+                    }}
+                >
+                    <MenuList
+                        sx={{
+                            mt: 0,
+                            '& .MuiMenuItem-root': {
+                                marginBottom: '1rem',
+                                '&:first-of-type': {
+                                    paddingTop: '4px',
+                                },
+                                '&:last-of-type': {
+                                    marginBottom: 0,
+                                },
+                            },
+                        }}
+                    >
+                        {tenantList}
+                    </MenuList>
+                </Drawer>
+            </>
+        );
+    }
+
+    return (
+        <DropdownMenu
```

Comment:
I believe this would also be a navigation menu since the user would be navigated away from their current page if they selected an option (as opposed to filling out a dropdown in a form). If that's the case, then I'd recommend we make this a `nav` element and label - which will allow it to show up in screen reader rotors.